src/OrchardCore/Orchard.Localization.Abstractions/StringLocalizerExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Extensions.Localization;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orchard.Localization.Abstractions
+{
+    public static class StringLocalizerExtensions
+    {
+        public static LocalizedString Plural(this IStringLocalizer localizer, string name, string pluralText, int count, params object[] arguments)
```

Comment:
It isn't possible to create a static indexer or to add an 'extension property' that returns an object implementing the indexer. There are several long threads in [Roslyn](https://github.com/dotnet/roslyn/issues/11159) and [C# lang](https://github.com/dotnet/csharplang/issues/192) repos regarding this topic, so it might be possible in the future. For now `Plural` stays an extension method.